### PR TITLE
Sort country list and add search in the intervention form

### PR DESCRIPTION
**Pull request by mariasola**
### General description

Adding sorting alphabetically and search functionality in the Country input in intervention form.

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

- In an Scenario, creating a new one.
- There, in step 2 there is a Country field. See if it is possible to Search and if it is ordered. 

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging

- [ ] Branch name / PR includes the related Jira ticket Id.
- [ ] Tests to check core implementation / bug fix added.
- [ ] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [ ] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: client/src/containers/interventions/new/step2/supplier/component.tsx [hunk 1]
```diff
@@ -85,6 +89,16 @@ const Supplier: FC = () => {
     [setValue, clearErrors],
   );
 
+  const handleDropdownSearch = useCallback((id: string, searchedCountry: string) => {
+      if (optionsCountries.some((country) => country['label'] === searchedCountry)) {
+        clearErrors(id);
+        setValue(id, searchedCountry);
+        console.log('SearchedCountry', searchedCountry);
```
Comment (davidsingal) [inline client/src/containers/interventions/new/step2/supplier/component.tsx:96]: Please remove this `console.log`

File: client/src/containers/interventions/new/step2/supplier/component.tsx [hunk 2]
```diff
@@ -85,6 +89,16 @@ const Supplier: FC = () => {
     [setValue, clearErrors],
   );
 
+  const handleDropdownSearch = useCallback((id: string, searchedCountry: string) => {
+      if (optionsCountries.some((country) => country['label'] === searchedCountry)) {
+        clearErrors(id);
```
Comment (davidsingal) [inline client/src/containers/interventions/new/step2/supplier/component.tsx:94]: for a search `clearErrors` is not needed

File: client/src/containers/interventions/new/step2/supplier/component.tsx [hunk 3]
```diff
@@ -173,13 +187,17 @@ const Supplier: FC = () => {
...
                 loading={isLoadingCountries}
                 current={optionsCountries.find(
                   (option) => option.value === watch('newLocationCountryInput'),
                 )}
                 options={optionsCountries}
                 placeholder="All Countries"
                 onChange={(value) => handleDropdown('newLocationCountryInput', value)}
+                onSearch={(searchedCountry) =>
+                  handleDropdownSearch('newLocationCountryInput', searchedCountry)
+                }
```
Comment (davidsingal) [inline client/src/containers/interventions/new/step2/supplier/component.tsx:200]: If I'm not wrong this `onSearch` is not needed, by default the search works. This is only needed when you want different behavior for the search.

**Review by davidsingal — CHANGES_REQUESTED** [review]

**Comment by davidsingal** [comment]
There is also an error in the compilation. Please, make sure your eslint extension is working well.

File: client/src/containers/interventions/new/step2/supplier/component.tsx (hunk 3 above)
Comment (mariasola) [reply client/src/containers/interventions/new/step2/supplier/component.tsx:200]: You are 100% right. Did not notice this, so now the `handleDropdownSearch ` is useless. Will remove and clean. Thanks!

**Review by davidsingal — APPROVED** [review]

**Pull request merged**